### PR TITLE
Adding API for connecting the MCU to a local WiFi network

### PR DESCRIPTION
**Pull request by aematei**
# Description

Creates an API for connecting the Pico to a WLAN. This includes:
- connecting/disconnecting to the network
- setting/getting connection SSID and password
- querying connection status
- ~saving connection credentials for easy re-connection later~ out of scope, moved  to a new ticket
- ~creating a new connection from an prior, saved connection~ out of scope, moved to a new ticket

Subsequent # LL-117

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Manual tests of methods directly on the MCU
- [x] Automated Unit Testing

# Checklist (when relevant):

- [X] My code follows the style guidelines of this project

[TypeScript](https://google.github.io/styleguide/tsguide.html)
[Python](https://peps.python.org/pep-0008/)
[Schema](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/forefront-2010/ee652242(v=vs.100))
- [X] I have included a reviewer on the pull request [Aaron and Nick]
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: controller/src/board/wlan.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+if "rp2" in os.uname():
+    import network # type: ignore pylint: disable=E0401
+    import rp2 # type: ignore pylint: disable=E0401
+
+else:
+    from controller.tests.mpstubs import network, rp2
+
+
+# constants
+_COUNTRY = 'US'
```
Comment (pogi7) [inline controller/src/board/wlan.py:15]: Our end users may not be in the `US`.  This constant should be able to change based on an input.

File: controller/src/board/wlan.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# wlan.py
+import time
+from .credentials import WLANCredentials
+import os
+
+if "rp2" in os.uname():
```
Comment (pogi7) [inline controller/src/board/wlan.py:6]: I cannot call this method from a windows computer.  

`os.name` works for me and performs the same function.

File: controller/tests/test_waln.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    wlan._wlan.connect_state = True
+    with pytest.raises(wlan.WLANConnectionActiveError):
+        wlan.connect(test_credentials)
+
+def test_disconnect():
+    wlan._wlan.connect_state = True
+    wlan.disconnect()
+    assert wlan._wlan.connect_state == False
+
+def test_getCredentials():
```
Comment (pogi7) [inline controller/tests/test_waln.py:49]: aaron@<computer> ~/git/luminosity-led (LL-117)
$ pytest controller/tests/test_waln.py 
====================================================================== test session starts =======================================================================
platform win32 -- Python 3.10.7, pytest-7.2.0, pluggy-1.0.0
rootdir: C:\Users\aaron\git\luminosity-led
plugins: mock-3.10.0
collected 9 items

controller\tests\test_waln.py ......F..                                                                                                                     [100%]

============================================================================ FAILURES ============================================================================ 
______________________________________________________________________ test_getCredentials _______________________________________________________________________

    def test_getCredentials():
>       assert isinstance(wlan.getCredentials(), credentials.WLANCredentials)
E       AssertionError: assert False
E        +  where False = isinstance(None, <class 'controller.src.board.credentials.WLANCredentials'>)
E        +    where None = <function getCredentials at 0x000001C9C5CF7370>()
E        +      where <function getCredentials at 0x000001C9C5CF7370> = wlan.getCredentials
E        +    and   <class 'controller.src.board.credentials.WLANCredentials'> = credentials.WLANCredentials

controller\tests\test_waln.py:50: AssertionError
==================================================================== short test summary info ===================================================================== 
FAILED controller/tests/test_waln.py::test_getCredentials - AssertionError: assert False
================================================================== 1 failed, 8 passed in 0.94s =================================================================== 
(.venv)

**Review by pogi7 — CHANGES_REQUESTED** [review]
@aematei please look at my comments.  how did you force push your changes?

It's usually not recommended to force-push changes unless you use force-push-with-lease

`git push --force-with-lease`

https://stackoverflow.com/questions/52823692/git-push-force-with-lease-vs-force

File: controller/src/board/wlan.py (hunk 2 above)
Comment (aematei) [reply controller/src/board/wlan.py:6]: Oh wow, yeah it looks like that call isn't supported on Windows. I'll try to find another way to do this. I essentially just want to find out whether or not to load stubs in place of the micropython libraries. Maybe I need to re-think the way I'm doing my unit testing in general.

I'll look for another way to do this in the meantime.

File: controller/src/board/wlan.py (hunk 1 above)
Comment (aematei) [reply controller/src/board/wlan.py:15]: Okay, sounds good. Where do you think the input will come from down the line? Will we be setting some initial variables before shipping the product out?

File: controller/tests/test_waln.py (hunk 3 above)
Comment (aematei) [reply controller/tests/test_waln.py:49]: Thanks for pointing this out, it should be a quick fix.

**Comment by aematei** [comment]
> please look at my comments. how did you force push your changes?

@pogi7 oh wow, I'm actually not sure. I was having some unexpected issues with trying to rebase my branch after pulling master. I got lost in git and tried to revert to my last commit. I think I need to do some more learning on how to keep my branch up to date with origin.

**Review by pogi7 — APPROVED** [review]
lgtm!
@aematei Feel free to merge

**Pull request merged**